cluster-autoscaler/cloudprovider/builder/cloud_provider_builder.go

```diff
@@ -47,70 +47,49 @@ var AvailableCloudProviders = []string{
 // DefaultCloudProvider is GCE.
 const DefaultCloudProvider = gce.ProviderNameGCE
 
-// CloudProviderBuilder builds a cloud provider from all the necessary parameters including the name of a cloud provider e.g. aws, gce
-// and the path to a config file
-type CloudProviderBuilder struct {
-	cloudProviderFlag       string
-	cloudConfig             string
-	clusterName             string
-	autoprovisioningEnabled bool
-	regional                bool
-}
-
-// NewCloudProviderBuilder builds a new builder from static settings
-func NewCloudProviderBuilder(cloudProviderFlag, cloudConfig, clusterName string, autoprovisioningEnabled, regional bool) CloudProviderBuilder {
-	return CloudProviderBuilder{
-		cloudProviderFlag:       cloudProviderFlag,
-		cloudConfig:             cloudConfig,
-		clusterName:             clusterName,
-		autoprovisioningEnabled: autoprovisioningEnabled,
-		regional:                regional,
-	}
-}
-
-// Build a cloud provider from static settings contained in the builder and dynamic settings passed via args
-func (b CloudProviderBuilder) Build(discoveryOpts cloudprovider.NodeGroupDiscoveryOptions, resourceLimiter *cloudprovider.ResourceLimiter) cloudprovider.CloudProvider {
-	glog.V(1).Infof("Building %s cloud provider.", b.cloudProviderFlag)
-	switch b.cloudProviderFlag {
+// NewCloudProvider builds a cloud provider from provided parameters.
+func NewCloudProvider(opts static.AutoscalingOptions, do cloudprovider.NodeGroupDiscoveryOptions, rl *cloudprovider.ResourceLimiter) cloudprovider.CloudProvider {
+	glog.V(1).Infof("Building %s cloud provider.", opts.CloudProviderName)
+	switch opts.CloudProviderName {
 	case gce.ProviderNameGCE:
```

Comment:
I'd rather keep the log above. It's 1 line / CA instance and it can be useful to quickly figure out cloudprovider from logs.

Reply:
Done